src/cml.js

```diff
@@ -510,10 +510,12 @@ class CML {
         fetchDepth = 0;
       }
     }
-    if (fetchDepth <= 0) {
-      return await exec('git fetch --all --unshallow');
+    if (fetchDepth !== undefined) {
```

Comment:
What would happen if `fetchDepth` is not passed/`undefined` now? It just wouldn't fetch anything?

Reply:
`fetchDepth` is not passed it will be undefined and nothing will happen.

Reply:
@deepyaman with your original code if a user did a GitHub actions `checkout.with.fetch-depth: 0` then ran a plain `cml ci` to say configure a git user then this would shallow their repo, something I didn't test for 🤷 . So now the default behavior with no options is the same as before, but a user can explicitly add a custom fetch-depth with their `cml ci` command

Reply:
Got it, thanks for the explanation!